packages/mobile/src/firebase/saga.ts

```diff
@@ -100,55 +100,62 @@ function celoGoldExchangeRateHistoryChannel(lastTimeUpdated: number) {
   }
 
   const now = Date.now()
+  // timestamp + 1 is used because .startAt is inclusive
+  const startAt = Math.max(lastTimeUpdated + 1, now - MAX_HISTORY_RETENTION)
 
   return eventChannel((emit: any) => {
-    const emitter = (snapshot: FirebaseDatabaseTypes.DataSnapshot) => {
+    const singleItemEmitter = (snapshot: FirebaseDatabaseTypes.DataSnapshot) => {
+      emit([snapshot.val()])
+    }
+    let cancelFunction = () => {}
+    const listenForNewElements = (newElementsStartAt: number) => {
+      cancelFunction = firebase
+        .database()
+        .ref(`${EXCHANGE_RATES}/cGLD/cUSD`)
+        .orderByChild('timestamp')
+        .startAt(newElementsStartAt)
+        .on('child_added', singleItemEmitter, errorCallback)
+    }
+    const fullListEmitter = (snapshot: FirebaseDatabaseTypes.DataSnapshot) => {
       const result: ExchangeRate[] = []
       snapshot.forEach((childSnapshot: FirebaseDatabaseTypes.DataSnapshot) => {
         result.push(childSnapshot.val())
         return undefined
       })
-      emit(result)
+      if (result.length) {
+        emit(result)
+        listenForNewElements(result[result.length - 1].timestamp + 1)
+      } else {
+        listenForNewElements(startAt)
+      }
     }
 
-    // timestamp + 1 is used because .startAt is inclusive
-    const startAt = lastTimeUpdated + 1 || now - MAX_HISTORY_RETENTION
-
-    const onValueChange = firebase
+    firebase
       .database()
       .ref(`${EXCHANGE_RATES}/cGLD/cUSD`)
       .orderByChild('timestamp')
       .startAt(startAt)
-      .on(VALUE_CHANGE_HOOK, emitter, errorCallback)
-
-    const cancel = () => {
-      firebase
-        .database()
-        .ref(`${EXCHANGE_RATES}/cGLD/cUSD`)
-        .orderByChild('timestamp')
-        .startAt(startAt)
-        .off(VALUE_CHANGE_HOOK, onValueChange)
-    }
+      .once(VALUE_CHANGE_HOOK, fullListEmitter, errorCallback)
```

Comment:
Unrelated to this PR but why is `VALUE_CHANGE_HOOK = 'value'`? I would assume it would equal `child_added`.

Reply:
If you pass `value` it returns the whole collection/object in one go. `child_added` is used so that the callback is called once for each child immediately and then whenever a new child is added it's called again. There's also a `child_updated`, `child_moved` and `child_removed`

Reply:
Right, so I find it confusing that `VALUE_CHANGE_HOOK` doesn't mean return values when there is a change but instead means return all values. Maybe just me though